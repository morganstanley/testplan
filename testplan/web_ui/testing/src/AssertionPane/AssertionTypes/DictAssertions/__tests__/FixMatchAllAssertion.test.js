import React from "react";
import { render } from "@testing-library/react";
import { StyleSheetTestUtils } from "aphrodite";

import DictMatchAllAssertion from "../DictMatchAllAssertion";

function defaultProps() {
  return {
    assertion: {
      category: "DEFAULT",
      description: "fix match all",
      key_weightings: null,
      matches: [
        {
          description: "fix match all 1/2: expected[1] vs values[0]",
          comparison: [
            [10914, "p", ["str", "c1dec2c5"], ["str", "c1dec2c5"]],
            [38, "p", ["str", "500"], ["str", "500"]],
            [44, "p", ["float", 9.0], ["str", "9"]]
          ],
          passed: true,
          comparison_index: 1
        },
        {
          description: "fix match all 2/2: expected[0] vs values[1]",
          comparison: [
            [10914, "p", ["REGEX", ".+"], ["str", "f3ea6276"]],
            [38, "p", ["int", 501], ["int", 501]],
            [44, "p", ["float", 9.1], ["float", 9.1]]
          ],
          passed: true,
          comparison_index: 0
        }
      ],
      meta_type: "assertion",
      passed: true,
      timestamp: 1747749737.592704,
      type: "FixMatchAll"
    },
  };
}

describe("DictMatchAllAssertion", () => {
  let component;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  it("renders the correct HTML structure", () => {
    component = render(<DictMatchAllAssertion {...defaultProps()} />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});