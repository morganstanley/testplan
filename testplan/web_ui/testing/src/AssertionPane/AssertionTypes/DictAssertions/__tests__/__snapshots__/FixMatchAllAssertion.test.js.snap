// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictMatchAllAssertion renders the correct HTML structure 1`] = `
<DocumentFragment>
  <div
    class="card_10lpuz2 card"
  >
    <div
      class="cardHeader_1s246z6-o_O-cardHeaderColorPassed_1nxwz58 card-header"
    >
      <div
        style="display: flex;"
      >
        <span
          class="button_13xlah4"
          style="order: 4; flex-grow: 4; padding: .125rem 0.75rem; display: flex; align-items: center;"
        >
          <span
            style="font-weight: bold;"
          >
            1/2: expected[1] vs values[0] 
          </span>
        </span>
        <span
          class="cardHeaderAlignRight_107ja4p"
        />
      </div>
    </div>
    <div
      class="collapseDiv_1cips5t collapse"
      style="padding-right: 1.25rem;"
    >
      <div
        class="assertionCardBody_1t2a0td card-body"
      />
    </div>
  </div>
  <div
    class="card_10lpuz2 card"
  >
    <div
      class="cardHeader_1s246z6-o_O-cardHeaderColorPassed_1nxwz58 card-header"
    >
      <div
        style="display: flex;"
      >
        <span
          class="button_13xlah4"
          style="order: 4; flex-grow: 4; padding: .125rem 0.75rem; display: flex; align-items: center;"
        >
          <span
            style="font-weight: bold;"
          >
            2/2: expected[0] vs values[1] 
          </span>
        </span>
        <span
          class="cardHeaderAlignRight_107ja4p"
        />
      </div>
    </div>
    <div
      class="collapseDiv_1cips5t collapse"
      style="padding-right: 1.25rem;"
    >
      <div
        class="assertionCardBody_1t2a0td card-body"
      />
    </div>
  </div>
</DocumentFragment>
`;
